Lib/test/test_httpservers.py

```diff
@@ -46,13 +51,23 @@ def read(self, n=None):
 
 
 class TestServerThread(threading.Thread):
-    def __init__(self, test_object, request_handler):
+    def __init__(self, test_object, request_handler, tls=None):
         threading.Thread.__init__(self)
         self.request_handler = request_handler
         self.test_object = test_object
+        self.tls = tls
 
     def run(self):
-        self.server = HTTPServer(('localhost', 0), self.request_handler)
+        if self.tls:
+            self.server = HTTPSServer(
+                ('localhost', 0),
+                self.request_handler,
+                certfile=self.tls[0],
+                keyfile=self.tls[1],
+                password=self.tls[2],
+            )
+        else:
```

Comment:
Ok this actually convinces me of the creating the following (function) helper:

```py

class DummyRequestHandler(NoLogRequestHandler, SimpleHTTPRequestHandler):
    pass

def create_https_server(
    certfile, 
    keyfile=None,
    password=None,
    *,
    address=('localhost', 0),
    request_handler=DummyRequestHandler,

):
    return HTTPSServer(
        address, request_handler, 
        certfile=certfile, keyfile=keyfile, password=password
    )
```

it will be cleaner like this, and in tests, we can forget about passing keywords
just to ease our life. WDYT?

Reply:
yea, let's do it